### PR TITLE
feat: project.dateModified in search results

### LINKED ISSUE #1564: Add `dateModified` to DS and Project
As a renku graph API user, I'd like the Cross-Entity Search and the Recently Viewed Entities APIs to return info about the modification date.

Acceptance criteria:
- [x] add `dateModified` to all the Project entities in the TS
  - [x] find out if `updated_at` returned by the Single Project GL API gets updated on a commit as well as GL Project metadata changes
  - [x] should we just rely on the mentioned `updated_at` and not even try to compute the date from the payload or perhaps, we'd need to do a max of: project `dateModified` (GL's `updated_at`), datasets' `dateModified`, activities' `startTime` and plans' `dateCreated`
  - [x] make sure relevant flows have the logic of inserting the `dateModified` into the TS:
    - [x] `TRIPLES_GENERATED`
    - [x] `ADD_MIN_PROJECT_INFO`
    - [x] `SYNC_REPO_METADATA`
      - [x] perhaps, we'd need to fetch `dateModified` from TS as an option and create an issue to make it mandatory in the changed flows code later? -> #1575 
  - [x] create a TS migration that adds all the data to TS ~(probably just the Projects graph)~
- [x] make the `GitLabProjectInfo` (project transformation flow) to calculate `dateModified` as max of (`updated_at`, `last_activity_at`, `created_at`)
- [x] go through all the code-base and remove the Option from all the places where Project dateModified is used
  - [x] make sure the `io.renku.graph.acceptancetests.data.Project` don't specify its own `dateModified` but uses the model's `dateModified`
- [x] make the Recently Viewed Entities return the `dateModified` for Project entities
- [x] make the Recently Viewed Entities return the `dateModified` for Dataset entities
- [x] make the Cross-Entity Search return the `dateModified` for Project entities
  - [x] the assumption is that the `date` in the response will still be pointing to `dateCreated` or we should maybe introduce a new `dateCreated` to all the entities and start phasing out the old ambiguous `date` property.
    - [x] if we decide to phase out the `date` we need to:
      - [ ] to talk to the UI and omnibenchmark teams asking to switch to the new `dateCreated`
      - [x] create a new issue to remove the old `date` from the responses
- [x] make the Project Details API return the `dateModified` from the TS (instead the `last_activity_at` from GL)
  - [x] `GitLabProject.dateModified` should go
- [x] make the Cross-Entity Search return the `dateModified` for Dataset entities
- [x] make the Cross-Entity Search to offer sorting by `dateModified`
- [x] after we have the new sortby=dateModified, create an issue to remove the sortBy=date

### PR DESCRIPTION
**Pull request by eikek**
Adopts our code for the fact that `project.dateModified` is mandatory. Adds the new data field to search and recently-viewed responses.

Refs: #1564

### CODE REVIEW

File: entities-search/src/main/scala/io/renku/entities/search/model.scala [hunk 1]
```diff
@@ -48,6 +48,7 @@ object model {
         name:             projects.Name,
         visibility:       projects.Visibility,
         date:             projects.DateCreated,
```
Comment (jachro) [inline entities-search/src/main/scala/io/renku/entities/search/model.scala:50]: We could also rename the `date` to `dateCreated` here or leave it for the next PR. Not sure what would work better. Probably it doesn't matter much as the next PR is something we'll do soon anyway. Maybe just worth taking a note.

File: triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/syncrepometadata/processor/model.scala [hunk 2]
```diff
@@ -23,6 +23,8 @@ import io.renku.graph.model.images.{Image, ImageUri}
 import io.renku.graph.model.projects
 import io.renku.triplesstore.SparqlQuery
 
+import java.time.Instant
+
 private sealed trait DataExtract {
   val path:              projects.Path
   val name:              projects.Name
```
Comment (jachro) [inline triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/syncrepometadata/processor/model.scala:30]: I'm adding a note here but actually, I wanted to put it against the `maybeDateModified` val which is one line below. Should we maybe unwrap it, too, I mean take the Option out of it?

**Review by jachro — APPROVED** [review]

File: triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/syncrepometadata/processor/model.scala (hunk 2 above)
Comment (eikek) [reply triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/syncrepometadata/processor/model.scala:30]: Yes, sure if that is possible? I wasn't sure what is considered "required" in the response from GL.

**Comment by eikek** [comment]
I'm addressing the comments in a next PR, so this can be merged :)

File: triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/syncrepometadata/processor/model.scala (hunk 2 above)
Comment (jachro) [reply triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/syncrepometadata/processor/model.scala:30]: Ah, yes, you are right. I think we talked about that.

**Pull request merged**